### ISSUE TITLE
DL4006 make false suggestions

### ISSUE THREAD

**Issue by j75** [hadolint/hadolint#522]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
No errors in Dockerfile.

### Actual behavior
`DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it. `
However, if I add
```
set -o pipefail
```
it now has an additional message:  `SC2039 In POSIX sh, set option pipefail is undefined.`

### Steps to reproduce the behavior
My Dockerfile is attached.

```bash
Haskell Dockerfile Linter v1.19.0-0-g94f073c
```

```Dockerfile
ARG  GHOSTSCR9_VERS=53.3
ARG  GHOSTSCR9_DIR_VERS=533
ARG  URL_GS=https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs9${GHOSTSCR9_DIR_VERS}

FROM centos:7.9.2009 AS centos_base

ARG  GHOSTSCR9_VERS
ARG  GHOSTSCR9_DIR_VERS
ARG  URL_GS

RUN  set -o pipefail; curl -s -L ${URL_GS}/ghostxps-9.${GHOSTSCR9_VERS}-linux-x86_64.tgz | tar -C /tmp -xzf - && \
        mv /tmp/ghostxps-9.${GHOSTSCR9_VERS}-linux-x86_64/gxps-9${GHOSTSCR9_DIR_VERS}-linux-x86_64 /usr/local/bin/gxps && \
        rm -rf /tmp/ghostxps-9.${GHOSTSCR9_VERS}-linux-x86_64/ && \
        chown root:root /usr/local/bin/g* && \
    ls -alF /usr/local/bin/g* 

ENTRYPOINT [ "/usr/bin/bash" ]

```
Ubuntu 20.04.1 with 5.4.0-65-generic kernel.

**Comment by lorenzo** [comment]
Please check the wiki page for this error for a better fix
